firefox-ios/firefox-ios-tests/Tests/ClientTests/TabTray/InactiveTabsTelemetryTests.swift

```diff
@@ -9,49 +9,73 @@ import XCTest
 
 final class InactiveTabsTelemetryTests: XCTestCase {
     var subject: InactiveTabsTelemetry?
+    var gleanWrapper: MockGleanWrapper!
 
     override func setUp() {
         super.setUp()
-        // Due to changes allow certain custom pings to implement their own opt-out
-        // independent of Glean, custom pings may need to be registered manually in
-        // tests in order to puth them in a state in which they can collect data.
-        Glean.shared.registerPings(GleanMetrics.Pings.shared)
-        Glean.shared.resetGlean(clearStores: true)
-        subject = InactiveTabsTelemetry()
+        gleanWrapper = MockGleanWrapper()
+        subject = InactiveTabsTelemetry(gleanWrapper: gleanWrapper)
     }
 
     override func tearDown() {
         subject = nil
+        gleanWrapper = nil
         super.tearDown()
     }
 
     func testRecordInactiveTabWhenSectionShownThenGleanIsCalled() throws {
         subject?.sectionShown()
 
-        testCounterMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.inactiveTabShown)
+        XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
+        if let savedMetric = gleanWrapper.savedEvent as? CounterMetricType {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabShown))
+        } else {
+            XCTFail("savedEvent is not of type CounterMetricType")
+        }
```

Comment:
to avoid nesting and easier readability, what do you think about the following?
```suggestion
let savedMetric = try XCTUnwrap(gleanWrapper.savedEvent as? CounterMetricType)
XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabShown))
XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
```